core/src/consensus/tendermint/network.rs

```diff
@@ -154,19 +154,18 @@ impl TendermintExtension {
         }
     }
 
-    fn request_proposal_to_any(&self, round: SortitionRound) {
-        for (token, peer) in &self.peers {
-            let is_future_height_and_view = round < peer.vote_step.into();
+    fn request_proposal_to_superiors(&self, round: SortitionRound, my_highest: Option<Priority>) {
+        if let Some(peer_highest_priority) = self.peers.iter().map(|(_id, peer)| peer.priority()).max() {
+            let highest_priority_peers =
+                self.peers.iter().filter(|(_id, peer)| peer.priority() == peer_highest_priority);
 
-            if is_future_height_and_view {
-                self.request_proposal(token, round);
-                continue
-            }
-
-            let is_same_height_and_view = round == peer.vote_step.into();
+            for (token, peer) in highest_priority_peers {
+                let is_future_height_and_view = round < peer.vote_step.into();
+                let is_same_height_and_view = round == peer.vote_step.into();
 
-            if is_same_height_and_view && peer.proposal.is_some() {
-                self.request_proposal(token, round);
+                if is_future_height_and_view || (is_same_height_and_view && peer.priority() > my_highest) {
+                    self.request_proposal(token, round);
+                }
```

Comment:
@HoOngEe This logic seems strange. IMO, comparing priorities for different rounds has no meaning. Could you explain rationale in this code to me?

Reply:
Yes it's weird. I'll modified the code to request to future peers or current round highest peers